holoviews/core/overlay.py

```diff
@@ -52,7 +53,7 @@ class CompositeOverlay(ViewableElement, Composable):
     _deep_indexable = True
 
     def hist(self, dimension=None, num_bins=20, bin_range=None,
-             adjoin=True, index=0, **kwargs):
+             adjoin=True, index=None, **kwargs):
```

Comment:
I kept the `index` functionality, however I decided to change the default behavior to show all histograms (which I believe to be a better default if we have the option).